### PR TITLE
[BOOTVID] Add boot video driver for NEC PC-98 series

### PR DESCRIPTION
**Pull request by disean**
## Purpose

Add missing driver.

AFTER:
bv

### CODE REVIEW

File: drivers/base/bootvid/i386/pc98/bootvid.c [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+ * COPYRIGHT:   Copyright 2020 Dmitry Borisov (di.sean@protonmail.com)
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include "precomp.h"
+
+/* GLOBALS ********************************************************************/
+
+static ULONG_PTR VideoMemoryI;
```
Comment (HBelusca) [inline drivers/base/bootvid/i386/pc98/bootvid.c:14]: What does the final "I" mean?

File: drivers/base/bootvid/i386/pc98/bootvid.c [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+    if (!FrameBuffer) MmUnmapIoSpace((PVOID)FrameBuffer, PEGC_FRAMEBUFFER_SIZE);
+
+    return FALSE;
+}
+
+VOID
+NTAPI
+VidCleanUp(VOID)
+{
+    NOTHING;
```
Comment (HBelusca) [inline drivers/base/bootvid/i386/pc98/bootvid.c:382]: No cleanup? (calling VidResetDisplay and unmapping memory)

File: drivers/base/bootvid/bootvid.rc [hunk 3]
```diff
@@ -1,5 +1,11 @@
 #define REACTOS_VERSION_DLL
+
+#if defined(SARCH_PC98)
+#define REACTOS_STR_FILE_DESCRIPTION  "NEC PC-98 Video Boot Driver"
```
Comment (binarymaster) [inline drivers/base/bootvid/bootvid.rc:4]:
```suggestion
#define REACTOS_STR_FILE_DESCRIPTION  "NEC PC-98 Boot Video Driver"
```

**Comment by HBelusca** [comment]
@disean : do you have a screenshot for demo purposes? :)

File: drivers/base/bootvid/i386/pc98/bootvid.c (hunk 1 above)
Comment (disean) [reply drivers/base/bootvid/i386/pc98/bootvid.c:14]: Intensity plane.

File: drivers/base/bootvid/i386/pc98/bootvid.c (hunk 2 above)
Comment (disean) [reply drivers/base/bootvid/i386/pc98/bootvid.c:382]: Not sure. The `VidInitialize` function is called exactly once. What happens if you press the `TAB+K` multiple times?

Comment (HBelusca) [reply drivers/base/bootvid/i386/pc98/bootvid.c:382]: Check what kind of things the regular VGA boot driver does.

File: drivers/base/bootvid/i386/pc98/bootvid.c [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+    _In_ ULONG Count)
+{
+    ULONG i;
+    PULONG Entry = Table;
+
+    for (i = 0; i < Count; i++, Entry++)
+        SetPaletteEntryRGB(i, *Entry);
+
+    for (i = Count; i < PEGC_MAX_COLORS; i++)
+        SetPaletteEntryRGB(i, 0);
```
Comment (binarymaster) [inline drivers/base/bootvid/i386/pc98/bootvid.c:251]:
```suggestion
        SetPaletteEntryRGB(i, VidpDefaultPalette[BV_COLOR_BLACK]);
```

**Comment by Saibamen** [comment]
What is the purpose of working on this old, not used and japanese-market only NEC PC-98 on ROS?

**Comment by binarymaster** [comment]
@Saibamen GitHub PR is not right place for such discussions, use the chat.

**Pull request merged**